### PR TITLE
Check native function declarations

### PR DESCRIPTION
**Pull request by turbolent**
## Description

Previously, the parser gained support for parsing the `native` modifier for functions, and the checker rejected the modifier, gated behind a configuration option.

Complete support for checking native functions, by requiring that if native functions are enabled, the function body is empty, and ignore checking it. 

______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/sema/errors.go [hunk 1]
```diff
@@ -878,6 +878,23 @@ func (e *InvalidNativeModifierError) Error() string {
...
+
+var _ SemanticError = &NativeFunctionWithImplementationError{}
+var _ errors.UserError = &NativeFunctionWithImplementationError{}
+
+func (*NativeFunctionWithImplementationError) isSemanticError() {}
+
+func (*NativeFunctionWithImplementationError) IsUserError() {}
+
+func (e *NativeFunctionWithImplementationError) Error() string {
+	return "native function may not have an implementation"
```
Comment (SupunS) [inline runtime/sema/errors.go:895]:
```suggestion
	return "native function must not have an implementation"
```

File: runtime/tests/checker/function_test.go
Comment (SupunS) [inline runtime/tests/checker/function_test.go:1]: Can you please also add a test to see if the composite functions get defined inside the composite type as members?

Comment (turbolent) [reply runtime/tests/checker/function_test.go:1]: Good idea! Added in d174f7be8598456f5c126c11ccb742b193208391

**Review by SupunS — APPROVED** [review]
Nice!

**Pull request merged**